tests/e2e/idps_test.go

```diff
@@ -0,0 +1,77 @@
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	conn "github.com/terraform-redhat/terraform-provider-rhcs/tests/ci"
+	"github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/cms"
+	con "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
+	exe "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/exec"
+)
+
+var _ = Describe("TF Test", func() {
+	Describe("Identity Providers test cases", func() {
+		var clusterID string
+		var idpServices map[string]*exe.IDPService
+		var (
+			htpasswdUsername = "jacko"
+			htpasswdPassword = "1q2wFe4rpoe2318"
+		)
+		htpasswdMap := []interface{}{map[string]string{"username": htpasswdUsername, "password": htpasswdPassword}}
+
+		BeforeEach(func() {
+			clusterID = os.Getenv("CLUSTER_ID")
+			// for future use
+			idpServices = map[string]*exe.IDPService{
+				"htpasswd": exe.NewIDPService(con.HtpasswdDir),
+				"github":   exe.NewIDPService(con.GithubDir),
+				"gitlab":   exe.NewIDPService(con.GitlabDir),
+				"google":   exe.NewIDPService(con.GoogleDir),
+				"ldap":     exe.NewIDPService(con.LdapDir),
+				"openid":   exe.NewIDPService(con.OpenidDir),
+			}
+		})
+		AfterEach(func() {
+			err := idpServices["htpasswd"].Destroy()
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Context("Author:smiron-High-OCP-63151 @OCP-63151 @smiron", func() {
+			It("OCP-63151 - Provision HTPASSWD IDP against cluster using TF", func() {
```

Comment:
Perhaps a context for each IDP would be simpler to grasp. I assume this file will grow, so containing only needed variables in a context can make lives easier down the road.

For example, you can have a BeforeEach in a context of HTPasswd that only creates an HTTPasswd IDP service, instead of creating all of them all the time.

Reply:
Oh i see you are using `By` 👍🏽 . Also a good way to do it. Though the variable initializations point is still valid IMO

Reply:
So I changed the structure a little, used `Describe` for each IDP inside the general Describe and 
initialized the vars in the `BeforeEach`. let me know what you think